libs/base/System.idr

```diff
@@ -182,6 +182,36 @@ namespace Escaped
   run : HasIO io => (cmd : List String) -> io (String, Int)
   run = run . escapeCmd
 
+||| Run a shell command, allowing processing its stdout line by line.
+|||
+||| Notice that is the line of the command ends with a newline character,
+||| it will be present in the string passed to the processing function.
+|||
+||| This function returns an exit code which value should be consistent with the `run` function.
+export
+covering
+runProcessingOutput : HasIO io => (String -> io ()) -> (cmd : String) -> io Int
+runProcessingOutput pr cmd = do
+  Right f <- popen cmd Read
+    | Left err => pure 1
+  True <- process f
+    | False => pure 1 -- we do not close `f` in case of reading error, line `run` does
```

Comment:
I don't understand this comment

Reply:
In the original `run` function `fclose f` is called only then `fRead f` returns `Right _`. I thought this was done intentionally (I don't know for sure the rules of when `fclose` should be called or not) and I repeated the same behaviour for this function, i.e. if any file error raises *during* the processing of the output of the spawned command, `fclose` is not called for `f`.

Reply:
Ah, `like` not `line

Reply:
oops